test/cluster-tests/kubernetes/test.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# exit on errors
+set -e
+# verbose logging
+set -x
+
+EXPECTED_NODE_COUNT=4
+EXPECTED_DNS=2
+
+namespace="namespace-${RANDOM}"
+echo "Running test in namespace: ${namespace}"
+trap teardown EXIT
+
+function teardown {
+  kubectl delete namespaces ${namespace}
+}
+
+
+echo "Testing number of nodes is ${EXPECTED_NODE_COUNT}"
```

Comment:
Host OS Upgrades or Service healing may happen from time to time so we may not have all nodes.  Should this retry for up to 10 minutes if not found?